jbmc/src/jbmc/jbmc_parse_options.cpp

```diff
@@ -243,13 +243,10 @@ void jbmc_parse_optionst::get_command_line_options(optionst &options)
   }
 
   if(!cmdline.isset("no-refine-strings"))
-  {
     options.set_option("refine-strings", true);
-    options.set_option("string-printable", cmdline.isset("string-printable"));
-    if(cmdline.isset("string-max-length"))
-      options.set_option(
-        "string-max-length", cmdline.get_value("string-max-length"));
-  }
+
+  if(cmdline.isset("string-printable"))
+    options.set_option("string-printable", true);
```

Comment:
I assume this is a deliberate change, that `string-printable` can now be set even if strings aren't being refined

Reply:
It shouldn't have any effect in that case. However, you're right it would make sense to check that string options are not used when strings are turned off.